File: javascript/tests/p_1079.test.ts
import { numTilePossibilities } from "../src/problems/p_1079";

describe("", () => {
  test("LeetCode 1079 - Letter Tile Possibilities", () => {
    expect(numTilePossibilities("AAB")).toBe(8);
    expect(numTilePossibilities("ABC")).toBe(15);
    expect(numTilePossibilities("A")).toBe(1);
  
    console.log("✅ All test cases passed!");
  });
})

File: golang/problems/p_1079_test.go
package problems

import (
	"testing"
)

func TestNumTilePossibilities(t *testing.T) {
	aab := NumTilePossibilities("AAB")
	abc := NumTilePossibilities("ABC")
	a := NumTilePossibilities("A")

	if aab != 8 {
		t.Errorf("AAB = %d; want 8", aab)
	}

	if abc != 15 {
		t.Errorf("ABC = %d; want 15", abc)
	}

	if a != 1 {
		t.Errorf("A = %d; want 1", a)
	}
}
